http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/NonJsonPayLoadIT.java

```diff
@@ -0,0 +1,176 @@
+            "    - action: \"POST\"\n" +
+            "      endpoint: \"/media-type/yaml\"\n" +
+            "      requestHeaders:\n" +
+            "        Content-Type: \"application/yaml\"\n" +
+            "      requestBody:\n" +
+            "        cities:\n" +
+            "          - name: Tokio\n" +
+            "            country: Japan\n" +
+            "  expectedResponse:\n" +
+            "    status: 200\n" +
+            "    headers:\n" +
+            "      Content-Type: \"application/json\"\n" +
+            "    body:\n" +
+            "      cities:\n" +
+            "        - name: Tokio\n" +
+            "          country: Japan\n";
+
+    private final static String EXPECTED_RESPONSE_BODY_YAML_WINDOWS_OS = "- description: \"Send a request with a YAML payload to the endpoint, including cities data\"\r\n"
+            +
+            "  steps:\r\n" +
+            "    - action: \"POST\"\r\n" +
+            "      endpoint: \"/media-type/yaml\"\r\n" +
+            "      requestHeaders:\r\n" +
+            "        Content-Type: \"application/yaml\"\r\n" +
+            "      requestBody:\r\n" +
+            "        cities:\r\n" +
+            "          - name: Tokio\r\n" +
+            "            country: Japan\r\n" +
+            "  expectedResponse:\r\n" +
+            "    status: 200\r\n" +
+            "    headers:\r\n" +
+            "      Content-Type: \"application/json\"\r\n" +
+            "    body:\r\n" +
+            "      cities:\r\n" +
+            "        - name: Tokio\r\n" +
+            "          country: Japan\r\n";
+
+    @Test
+    public void testExpectedXmlResponse() {
+        app.given()
+                .get("/city").then()
+                .statusCode(HttpStatus.SC_OK)
+                .contentType(MediaType.APPLICATION_XML)
+                .body("cityListWrapper.cities[0].name", equalTo("San Bernardino"))
+                .body("cityListWrapper.cities[0].country", equalTo("EEUU"));
+
+    }
+
+    @Test
+    public void testXMLPayloadPostRequest() throws JAXBException {
+
+        City city = new City("Paris", "France");
+        List<City> cityList = new ArrayList<>();
+        cityList.add(city);
+
+        CityListDTO requestCityList = new CityListDTO(cityList);
+
+        String xmlPayload = convertCityListDTOToXml(requestCityList);
+
+        String payload = app.given().contentType(MediaType.APPLICATION_XML)
+                .body(xmlPayload) // Use the generated XML payload
+                .when()
+                .post("/city")
+                .then()
+                .extract().asString();
+        Document doc = Jsoup.parse(payload);
+        String unescapedXml = doc.text();
+        assertThat(unescapedXml, equalTo(EXPECTED_XML_PAYLOAD));
+    }
+
+    @Test
+    public void testExpectedYamlResponse() {
+
+        final Response response = app.given()
+                .contentType("application/yaml")
+                .get("/city/getYamlFile")
```

Comment:
I'd like to point out that you don't actually test YAML. You test that you can send String from endpoint, consume string and because you manually serialized DTO to YAML, it is YAML.  The condition is always true because you do it manually.

Reply:
`testExpectedYamlResponse` is acceptable now. I happen to disagree